browser/ui/views/infobars/brave_global_infobar_manager.cc

```diff
@@ -0,0 +1,192 @@
+#include <vector>
+
+#include "brave/browser/ui/views/infobars/brave_confirm_infobar.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "components/infobars/content/content_infobar_manager.h"
+#include "components/infobars/core/infobar.h"
+#include "components/infobars/core/infobar_manager.h"
+
+namespace {
+std::unique_ptr<infobars::InfoBar> CreateBraveGlobalInfoBar(
+    std::unique_ptr<BraveGlobalConfirmInfobarDelegate> delegate) {
+  return std::make_unique<BraveConfirmInfoBar>(std::move(delegate));
+}
+
+content::WebContents* GetCurrentWebContents() {
+  Browser* browser = chrome::FindLastActive();
+  if (!browser) {
+    return nullptr;
+  }
+
+  TabStripModel* model = browser->tab_strip_model();
+  if (!model) {
+    return nullptr;
+  }
+
+  return model->GetActiveWebContents();
+}
+
+infobars::InfoBarManager* GetCurrentInfoBarManager() {
+  auto* web_contents = GetCurrentWebContents();
+  if (!web_contents) {
+    return nullptr;
+  }
+
+  infobars::ContentInfoBarManager* infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(web_contents);
+  if (!infobar_manager) {
+    return nullptr;
+  }
+
+  return infobar_manager;
+}
+
+void RemoveInfobarsByIdentifier(
+    infobars::InfoBarManager* infobar_manager,
+    const infobars::InfoBarDelegate::InfoBarIdentifier& id,
+    BraveGlobalInfoBarManager* observer) {
+  for (size_t i = 0; i < infobar_manager->infobar_count(); i++) {
+    if (auto* current_infobar = infobar_manager->infobar_at(i);
+        current_infobar->delegate()->GetIdentifier() == id) {
+      static_cast<BraveGlobalConfirmInfobarDelegate*>(
+          current_infobar->delegate())
+          ->RemoveObserver(observer);
+      infobar_manager->RemoveInfoBar(current_infobar);
+    }
+  }
+}
+
+void RemoveObserverForInfobarManagers(
+    const infobars::InfoBarDelegate::InfoBarIdentifier& id,
+    BraveGlobalInfoBarManager* observer) {
+  for (const auto* browser : *BrowserList::GetInstance()) {
+    const auto* tab_strip_model = browser->tab_strip_model();
+    for (int i = 0; i < tab_strip_model->count(); ++i) {
+      auto* web_contents = tab_strip_model->GetWebContentsAt(i);
+      DCHECK(web_contents);
+      if (!web_contents) {
+        continue;
+      }
+      if (auto* infobar_manager =
+              infobars::ContentInfoBarManager::FromWebContents(web_contents)) {
+        infobar_manager->RemoveObserver(observer);
+        RemoveInfobarsByIdentifier(infobar_manager, id, observer);
+      }
+    }
+  }
+}
+}  // namespace
+
+BraveGlobalInfoBarManager::BraveGlobalInfoBarManager(
+    std::unique_ptr<BraveGlobalConfirmInfoBarDelegateFactory> delegate_factory)
+    : delegate_factory_(std::move(delegate_factory)) {}
+
+BraveGlobalInfoBarManager::~BraveGlobalInfoBarManager() {
+  RemoveObserverForInfobarManagers(delegate_factory_->GetInfoBarIdentifier(),
```

Comment:
This should be in the OnManagerShuttingDownCall

Reply:
removed OnManagerShuttingDown at all